Add styled-components-ssr example

madebynoam opened the pull request:
Created a styled-components-ssr example for future reference. 

It includes a styled-components middleware for Tux.

madebynoam committed:
Add styled-components-ssr example

eirikurn commented:
This is great! Our plan though is to publish some of the more
common middlewares as official addons in the addon folder. Can
you look into doing that?


--
Eiríkur Heiðar Nilsson

+354 869 3179 / @eirikurn


On July 4, 2018 at 12:03 PM, Noam Almosnino
(notifications@github.com) wrote:

Created a styled-components-ssr example for future reference.
It includes a styled-components middleware for Tux.
You can view, comment on, or merge this pull request online at:
https://github.com/aranja/tux/pull/155
Commit Summary
* Add styled-components-ssr example

File Changes
* A examples/.DS_Store 
( https://github.com/aranja/tux/pull/155/files#diff-0 ) (0)

* A examples/styled-components-ssr/.gitignore 
( https://github.com/aranja/tux/pull/155/files#diff-1 ) (15)

* A examples/styled-components-ssr/.neutrinorc.js 
( https://github.com/aranja/tux/pull/155/files#diff-2 ) (12)

* A examples/styled-components-ssr/package.json 
( https://github.com/aranja/tux/pull/155/files#diff-3 ) (26)

* A examples/styled-components-ssr/src/app.js 
( https://github.com/aranja/tux/pull/155/files#diff-4 ) (15)

* A examples/styled-components-ssr/src/components/Description.js
( https://github.com/aranja/tux/pull/155/files#diff-5 ) (11)

* A examples/styled-components-ssr/src/components/Header.js 
( https://github.com/aranja/tux/pull/155/files#diff-6 ) (23)

* A examples/styled-components-ssr/src/components/Home.js 
( https://github.com/aranja/tux/pull/155/files#diff-7 ) (38)

* A
examples/styled-components-ssr/src/components/TuxLogo/TuxLogo.js
( https://github.com/aranja/tux/pull/155/files#diff-8 ) (17)

* A
examples/styled-components-ssr/src/components/TuxLogo/index.js 
( https://github.com/aranja/tux/pull/155/files#diff-9 ) (1)

* A examples/styled-components-ssr/src/components/TuxLogo/tux.svg
( https://github.com/aranja/tux/pull/155/files#diff-10 ) (5)

* A
examples/styled-components-ssr/src/middlewares/styled-components/index.js
( https://github.com/aranja/tux/pull/155/files#diff-11 ) (17)

* A
examples/styled-components-ssr/src/middlewares/styled-components/neutrino.js
( https://github.com/aranja/tux/pull/155/files#diff-12 ) (25)

* A examples/styled-components-ssr/src/server/index.js 
( https://github.com/aranja/tux/pull/155/files#diff-13 ) (16)

Patch Links:
* https://github.com/aranja/tux/pull/155.patch

* https://github.com/aranja/tux/pull/155.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub 
( https://github.com/aranja/tux/pull/155 ), or mute the thread 
( https://github.com/notifications/unsubscribe-auth/AAHBltJcluD5CBsWoILEm3j5aCok7eMpks5uDK7_gaJpZM4VCd-w )
.

madebynoam commented:
Thanks! Makes sense - I'll look into the addons next.

madebynoam committed:
Move middleware from example to addons folder. The example now uses this middleware.

Todo: Figure out the best way to test this middleware and best way to test styled components in general.

madebynoam commented:
Just committed the addon. The example now uses it as well. 
The remaining task is to figure out a good way to write a test for this addon and styled components, any thoughts?

eirikurn commented on addons/styled-components/package.json:
```diff
@@ -0,0 +1,31 @@
+  },
+  "peerDependencies": {
+    "react": ">16.0.0-beta",
+    "react-chain": "^0.5.0"
```
Need peer dependency for styled-components. Actually does not depend on react.

eirikurn replied:
It also has a dependency on @neutrinojs/compile-loader and peer dependency on neutrino.

eirikurn commented on .gitignore:
```diff
@@ -10,3 +10,9 @@ node_modules
+yarn.lock
+yarn-error.log
+yarn.lock
+yarn.lock
```
Duplicate ignore entries.

.DS_Store should arguably be in your [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

eirikurn reviewed (changes requested):
Awesome.

Fix the minor issues. Otherwise LGTM.

Would be great to get a basic README.md, can work from the react-router addon readme, but happy to get your input on documentation.

madebynoam commented:
Thanks! I'll be making the edits. Agree about the README.md, I'll work on it as well.
